pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -76,6 +76,11 @@ func (srv PrometheusSrv) RouteGetRuleStatuses(c *models.ReqContext) response.Res
 		return ErrResp(http.StatusInternalServerError, err, "failed to get namespaces visible to the user")
 	}
 
+	if len(namespaceMap) == 0 {
+		srv.log.Debug("User does not have access to any of the namespaces")
+		return response.JSON(http.StatusOK, ruleResponse)
```

Comment:
Do we want to return `HTTP 200 OK` if the user does not have access to at least one namespace?

Reply:
I believe so - returning something like `400` or `404` doesn't look right in my eyes as this doesn't necessarily mean an unexpected (or erroring) state?